lib/mayaHydra/hydraExtensions/sceneIndex/registration.cpp

```diff
@@ -417,10 +417,22 @@ MayaHydraSceneIndexRegistry::~MayaHydraSceneIndexRegistry()
         MSceneMessage::removeCallback(_AfterOpenCBId);
     }
     _AfterOpenCBId = 0;
+    _RemoveAllSceneIndexNodes();
```

Comment:
A ticket has been created to add a unit test once a mtoa build is ready in our automation suite.